src/Sylius/Bundle/LocaleBundle/Form/Extension/DateExtension.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\LocaleBundle\Form\Extension;
+
+use Sylius\Bundle\LocaleBundle\Form\DataTransformer\ArrayGregorianToCalendarSystemTransformer;
+use Sylius\Bundle\LocaleBundle\Form\DataTransformer\StringGregorianToCalendarSystemTransformer;
+use Sylius\Bundle\LocaleBundle\Templating\Helper\LocaleHelper;
+use Sylius\Component\Locale\Calendars;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\AbstractTypeExtension;
+
+/**
+ * Extension for Symfony's DateType for a localized calendar-aware date.
+ *
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
+ */
+class DateExtension extends AbstractTypeExtension
+{
+    /**
+     * @var LocaleHelper
+     */
+    protected $localeHelper;
+
+    /**
+     * @param LocaleHelper $helper
+     */
+    public function __construct(LocaleHelper $helper)
+    {
+        $this->localeHelper = $helper;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if (Calendars::GREGORIAN === $this->localeHelper->getCurrentCalendar()) {
+            // We only need to alter default behaviour when
+            // we're having a traditional calendar system.
+            return;
+        }
+
+        if ('single_text' === $options['widget']) {
+            $builder->addViewTransformer(new StringGregorianToCalendarSystemTransformer(
+                'Y-m-d',
+                $this->localeHelper
+            ));
+        } elseif ('array' === $options['input']) {
+            $builder->addViewTransformer(new ArrayGregorianToCalendarSystemTransformer(
+                $this->localeHelper
+            ));
+        }
```

Comment:
Why don't use just `addViewTransformer`? It will be totally clear what happen there.